lib/ExpressionUtils/core/exprMatrix_utils.py

```diff
@@ -203,3 +207,117 @@ def get_expression_matrix(self, params):
                                                   expr_set_data,
                                                   '{0}_TPM_ExpressionMatrix'.format(output_obj_name))
         return fpkm_ref, tpm_ref
+
+    # returns a list of [ min, max, mean, std.dev, is_data_missing] for one row of conditional 
+    # expression values
+
+    def get_matrix_stats( self, raw_row ):
+
+        #self.logger.info( "### gms raw row = {0}".format( pformat( raw_row ) ) )
+        has_missing = "No"
+        row = []
+        for r in raw_row:
+            if r == None or numpy.isnan( r ):     # careful here - r can be 0 which is a legitimate value
+                has_missing = "Yes"
+            else:
+                row.append(r)
+
+        if len( row ) < 1:
+            return( [ 'NA', 'NA', 'NA', 'NA', 'Yes' ] )
+        else:
+            if len( row ) == 1:
+               sd = 0
+            else:
+               sd = numpy.std( row, ddof=1 )
+            return( [ min( row ), max( row ), numpy.mean( row ), sd, has_missing ] )
+
+    # returns a dict that maps feature_id -> [ fc, q ]
+
+    def convert_dem_to_dict( self, dem ):
+
+        row_ids = dem.get( 'row_ids' )
+        vals = dem.get( 'values' )
+  
+        n_rows = len( row_ids )
+        if ( len( vals ) != n_rows ):
+            raise Exception( "length discrepancy in differential expression matrix: {0} row_ids but {1} values".format( n_rows, len( fvals ) ) )
+
+        dem_dict = {}
+        for i in range( 0, n_rows ):
+            dem_dict[ row_ids[i] ] = [ vals[i][0], vals[i][2] ] # [fc,q]. (not bothering to check for dups here)
+
+        return dem_dict
+
+    # implments get_enhancedFilteredExpressionMatrix() method
+
+    def get_enhancedFEM( self, params ):
+
+        if not params.get( 'fem_object_ref' ):
```

Comment:
Slightly clearer and more computationally efficient to say `if "fem_object_ref" not in params:`